Tasks/AzureKeyVault/operations/KeyVault.ts

```diff
@@ -162,4 +174,19 @@ export class KeyVault {
 
         return error;
     }
+
+    private generateProvisionKeyVaultPermissionsScript(): void {
+        let filePath = path.join(os.tmpdir(), "ProvisionKeyVaultPermissions.ps1");
+
```

Comment:
Would it be better to use the the _work/_temp directory instead of os temp directory. The reasoning behind this is that the agent tasks will have access to _work/_temp for sure.